### PR TITLE
Cleanup  : String.length() == 0 replaced with String.isEmpty, ...

### PR DESCRIPTION
**Pull request by doom369**
Motivation:

IDE warnings fixes.

Modification:

`String.length() == 0` replaced with `String.isEmpty()`, unnecessary assertion removed, unnecessary class cast and correct `StringBuilder` size, empty `else` block commented.

Result:

Code easier to read. No warnings.

### CODE REVIEW

File: codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java [hunk 1]
```diff
@@ -646,9 +647,10 @@ private LastHttpContent readTrailingHeaders(ByteBuf buffer) {
                         b.append(currentLastPos)
                          .append(lineTrimmed);
                         current.set(lastPos, b.toString());
-                    } else {
-                        // Content-Length, Transfer-Encoding, or Trailer
                     }
+                    //else {
+                    // Content-Length, Transfer-Encoding, or Trailer
```
Comment (Scottmitch) [inline codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java]: I realize you didn't introduce this comment but is this comment necessary? Can we make the comment more meaningful or just remove it?

@trustin - WDYT [1]

[1] https://github.com/doom369/netty/commit/4ce65f36212e7a97b902958ad6c56611091c83f5#diff-bba2e269877a9cc5f6b8ad204b1c89e5R451

Comment (trustin) [inline codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java]: I don't remember why I added that comment anymore unfortunately. I usually add a comment to an empty `else` block to explain when the condition of `if` clause will not match.

Comment (trustin) [inline codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java]: That being said, +1 for @Scottmitch's suggestion.

Comment (jasontedor) [inline codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java]: Knowing full well what I'm about to say will likely not have my intended effect, backfire and lead to me being viewed as contrary or difficult, as a user, follower, and occasional contributor to Netty, I feel it's worth mentioning these distractions (which happen quite frequently) just increase the cost of contributing. This comment has nothing to do with this PR. This discussion would have been better started on a separate issue.

Comment (normanmaurer) [inline codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java]: I agree with @jasontedor here... So all good @jasontedor  :)

Comment (normanmaurer) [inline codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java]: (also I will cleanup the stuff)

File: codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java [hunk 2]
```diff
@@ -646,9 +647,10 @@ private LastHttpContent readTrailingHeaders(ByteBuf buffer) {
                         b.append(currentLastPos)
                          .append(lineTrimmed);
                         current.set(lastPos, b.toString());
-                    } else {
-                        // Content-Length, Transfer-Encoding, or Trailer
                     }
+                    //else {
```
Comment (netkins) [inline codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java]: MAJOR This block of commented-out lines of code should be removed. [rule](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

File: codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java [hunk 3]
```diff
@@ -646,9 +647,10 @@ private LastHttpContent readTrailingHeaders(ByteBuf buffer) {
...
+                    // Content-Length, Transfer-Encoding, or Trailer
+                    //}
```
Comment (netkins) [inline codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java]: MAJOR This block of commented-out lines of code should be removed. [rule](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

**Comment by doom369** [comment]
I reverted this change so it could be fixed with another PR.

**Comment by normanmaurer** [comment]
Cherry-picked into 4.1 (506f0d8f8c10e1b24924f7d992a726d7bdd2e486) and 4.0 (49d4c35)

File: codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java (hunk 1 above)
Comment (Scottmitch) [inline codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java]: The intention is not to hold up the PR but as the code was modified it is a good opportunity to get input/justifications or things that smell fishy. Opening another issue sgtm (all good @jasontedor)

Comment (jasontedor) [inline codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java]: Thanks for listening @normanmaurer and @Scottmitch.

I understand where you're coming from @Scottmitch and I agree with your tacit point that this code should not have been changed in this PR in the first place as it has nothing to do with the stated purpose of the PR. My point is that for both contributors and maintainers a more optimal response would have been a request to revert and the creation of a new issue.

**Pull request closed without merging**